### ISSUE TITLE
Angular event types and docs 

### ISSUE THREAD

**Issue by razvangeangu** [ionic-team/stencil-ds-output-targets#54]
The outputs from the angular proxies should inherit the events from the stencil component class.

The current implementation adds the outputs as the properties of the angular class.

```ts
import { Components } from 'component-library'
export declare interface MyButton extends Components.MyButton {}
@ProxyCmp({ inputs: ["buttonType"] })
@Component({
  selector: "my-button",
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: "<ng-content></ng-content>",
  inputs: ["buttonType"],
})
export class MyButton {
  ionFocus!: EventEmitter<CustomEvent>;
  protected el: HTMLElement;
  constructor(c: ChangeDetectorRef, r: ElementRef, protected z: NgZone) {
    c.detach();
    this.el = r.nativeElement;
    proxyOutputs(this, this.el, ["ionFocus"]);
  }
}
```

I believe they should be inherited from the stencil class like this to inherit types and docs in similar fashion to how inputs are added.

```diff
import { Components } from 'component-library'
++ import { Button as IButton } from 'component-library/dist/types/components/my-button/my-button'
++ export declare interface MyButton extends Required<Pick<IButton, 'ionFocus'>> {}
export declare interface MyButton extends Components.MyButton {}
@ProxyCmp({ inputs: ["buttonType"] })
@Component({
  selector: "my-button",
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: "<ng-content></ng-content>",
  inputs: ["buttonType"],
++  outputs: ["ionFocus"]
})
export class MyButton {
--  ionFocus!: EventEmitter<CustomEvent>;
  protected el: HTMLElement;
  constructor(c: ChangeDetectorRef, r: ElementRef, protected z: NgZone) {
    c.detach();
    this.el = r.nativeElement;
    proxyOutputs(this, this.el, ["ionFocus"]);
  }
}
```

**Comment by jthoms1** [comment]
What would be the advantage of making this change. How would it change the experience for consumers of the components?

**Comment by razvangeangu** [comment]
When using the components in Angular it will point to the right interfaces and silent the errors.

<img width="942" alt="Screenshot 2020-06-19 at 23 04 18" src="https://user-images.githubusercontent.com/15063170/85182683-52ecd700-b281-11ea-9290-fdeefd5e574c.png">

**Comment by jthoms1** [comment]
Could you specify the ide you used for the screenshot?

**Comment by razvangeangu** [comment]
I am using webstorm for that screenshot.
